airbyte-config-oss/specs-oss/build.gradle

```diff
@@ -12,40 +12,39 @@ dependencies {
     implementation project(':airbyte-json-validation')
 }
 
-task generateSeedConnectorSpecs(type: JavaExec, dependsOn: compileJava) {
+task downloadConnectorRegistry(type: JavaExec, dependsOn: compileJava) {
+    /**
+     * run this once a day. if you want to force this task to run do so with --rerun
+     * e.g. ./gradlew :airbyte-config-oss:specs:downloadConnectorRegistry --info --rerun
+     */
+    inputs.property("todaysDate", new Date().clearTime() )
+    outputs.upToDateWhen { true }
     classpath = sourceSets.main.runtimeClasspath
-
-    mainClass = 'io.airbyte.configoss.specs.SeedConnectorSpecGenerator'
-
-    args '--seed-root'
-    args new File(project(":airbyte-config-oss:init-oss").projectDir, '/src/main/resources/seed')
+    mainClass = 'io.airbyte.configoss.specs.ConnectorRegistryDownloader'
+    args project(":airbyte-config-oss:init-oss").projectDir
 }
 
+project(":airbyte-config-oss:init-oss").tasks.processResources.dependsOn(downloadConnectorRegistry)
+project(":airbyte-config-oss:init-oss").tasks.processTestResources.dependsOn(downloadConnectorRegistry)
+project(":airbyte-config-oss:init-oss").tasks.test.dependsOn(downloadConnectorRegistry)
 
-task generateConnectorSpecsMask(type: JavaExec, dependsOn: generateSeedConnectorSpecs) {
-    classpath = sourceSets.main.runtimeClasspath
 
-    mainClass = 'io.airbyte.configoss.specs.ConnectorSpecMaskGenerator'
+task downloadConnectorSpecsMask(type: JavaExec, dependsOn: compileJava) {
+    /**
+     * run this once a day. if you want to force this task to run do so with --rerun
+     * e.g. ./gradlew :airbyte-config-oss:specs:downloadConnectorRegistry --info --rerun
+     */
+    inputs.property("todaysDate", new Date().clearTime() )
```

Comment:
I was modelling how we bring down the Connector Registry currently.

However in the registries case (for cloud) during run time we also continually poll for new updates using cron.

For the secret mask we do not do this, because to apply a new secret mask the whole app needs a restart.

Theres an issue on that concern here: [MaskedDataInterceptor is set at build time, should be updated at runtime - Research](https://app.zenhub.com/workspaces/connector-operations-6318ede5eda32e0010ef8566/issues/gh/airbytehq/airbyte/23254)